storage: de-flake TestStoreRangeRemoveDead

tbg opened the pull request:
The test was making a half-hearted attempt at not naming the leaseholder
the dead store, and was failing when those two coincided.

Hard-code the second store as the dead store and actually stop it (so
that the previous failure mode is now just a slower test run). This
also makes the test more realistic since now the dead node is actually
down.

Fixes #30708.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/30713)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

I'm not very familiar with this test, though. @spencerkimball might want to give this a thumbs up as well.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/30713#-:-LNQJ08y5XAfIqR5GapY:beqxja3)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

spencerkimball commented on pkg/storage/client_raft_test.go:
```diff
@@ -2900,9 +2904,15 @@ func TestStoreRangeRemoveDead(t *testing.T) {
+	// one since getRangeMetadata is hard-coded to query that one, so we'll use the
+	// one after that.
+	storeID := mtc.stores[1].StoreID() // 2
+	atomic.StoreInt32(&deadStoreID, int32(storeID))
```
No need to keep `deadStoreID` anymore. Remove this setting here and remove the variable declaration above.

spencerkimball replied:
You also don't need `storeID`.

tschottdorf replied:
Done, thanks!

tbg committed:
storage: de-flake TestStoreRangeRemoveDead

The test was making a half-hearted attempt at not naming the leaseholder
the dead store, and was failing when those two coincided.

Hard-code the second store as the dead store and actually stop it (so
that the previous failure mode is now just a slower test run). This
also makes the test more realistic since now the dead node is actually
down.

Fixes #30708.

Release note: None

tschottdorf commented:
Also fixed the data race in the test (mtc.stores gets mutated by stopStore).
